core/test_network/src/lib.rs

```diff
@@ -81,71 +75,43 @@ pub trait TestGenesis: Sized {
 impl TestGenesis for GenesisNetwork {
     fn test(submit_genesis: bool) -> Option<Self> {
         let cfg = Configuration::test();
-        let mut genesis = RawGenesisBlock::new(
-            "alice".parse().expect("Valid"),
-            "wonderland".parse().expect("Valid"),
-            get_key_pair().public_key().clone(),
-        );
+
+        // TODO: Fix this somehow. Probably we need to make `kagami` a library (#3253).
+        let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let content = std::fs::read_to_string(manifest_dir.join("../../configs/peer/genesis.json"))
```

Comment:
what would be appropriate is to use a function like `kagami::register_validator` to register all validators or to cherry pick required ones

Reply:
> what would be appropriate is to use a function like `kagami::register_validator`

Yes, that's why I created #3253. Although I'm not sure about `register_validator`. I thought about `generate_genesis`.

> Do you want to rely on example genesis.json in tests?

I don't know currently. With current approach we can be sure that our example `genesis.json` is pretty right and tested.
From the other side, we still add new instructions inside it, so it's note pure `geneis.json`.

I don't know if we should test `genesis.json` or test completely another genesis just with the same validators.

@appetrosyan ?

Reply:
I understand this is the reason for #3253 although I am not sure that it's appropriate solution for this issue and wanted to see what others think

Reply:
@appetrosyan

Reply:
Sorry, I was travelling, so this contributed to the delay. 

#3253 might be useful to resolve separately. Within the scope of this (already large) PR, it's more reasonable to recall that we want to get rid of `iroha_client::integration` tests, and not fix something which we might remove relatively soon. 

@astrokov7 's work on `iroha_clien_cli`-based tests should be very helpful in fixing the problem.

Reply:
> what would be appropriate is to use a function like kagami::register_validator to register all validators or to cherry pick required ones

I don't know if coupling our example crate to the testing framework is a good idea. I'm not sure that the marginal improvement in the DRY adherence is worth the extra overhead each time we need to run tests.

Reply:
> I don't know if coupling our example crate to the testing framework is a good idea. I'm not sure that the marginal improvement in the DRY adherence is worth the extra overhead each time we need to run tests.

I don't understand do you agree with me or not. I don't want to couple the example crate, i.e. kagami to the testing framework which is why I have started this discussion because we have couple our example genesis file with our testing and IMO these should stay separated. Anyhow, if this will be resolved with `iroha_client_cli` integration testing @Arjentix you can close this discussion